src/mlpack/core/data/split_data.hpp

```diff
@@ -224,6 +229,89 @@ Split(const arma::Mat<T>& input,
                          std::move(testData));
 }
 
+template <class fieldType,
+          class = std::enable_if_t<
+              arma::is_Col<typename fieldType::object_type>::value ||
+              arma::is_Mat_only<typename fieldType::object_type>::value>>
+void Split(fieldType& input, 
+           arma::field<arma::vec>& inputLabel,
+           fieldType& trainData,
+           arma::field<arma::vec>& trainLabels,
+           fieldType& testData, 
+           arma::field<arma::vec>& testLabels,
+           const double testRatio, 
+           const bool shuffleData = true) 
+{
+  const size_t testSize = static_cast<size_t>(input.n_cols * testRatio);
+  const size_t trainSize = input.n_cols - testSize;
+
+  arma::uvec order = arma::linspace<arma::uvec>(0, input.n_cols - 1,
+      input.n_cols);
+  if (shuffleData)
+    order = arma::shuffle(order);
+
+  if(trainSize > 0)
+  {
+    trainLabels.set_size(1, trainSize);
+
+    for(size_t i = 0; i < trainSize; i++)
+       trainData[i] = input(0,order(i));
+
+    // Field type has fixed size so we can't use span and assignment
+    // operator.
+    for (size_t i = 0; i < trainSize; i++)
+      trainLabels(0, i) = inputLabel[i];
+  }
+
+  if (testSize <= input.n_cols)
+  {
+    for(size_t i = trainSize; i < input.n_cols - 1; i++)
+       testData[i - trainSize] = input(0,order(i));
+    
+    testLabels.set_size(1, testSize);
+    for (size_t i = trainSize; i < input.n_cols; i++)
+      testLabels(0, i - trainSize) = inputLabel[i];
+  }
+  return;
+}
+
+template <class fieldType,
+          class = std::enable_if_t<
+              arma::is_Col<typename fieldType::object_type>::value ||
+              arma::is_Mat_only<typename fieldType::object_type>::value>>
+void Split(fieldType& input,
+          arma::field<arma::vec>& inputLabel,
+          fieldType&  trainData,
+          arma::mat& trainLabel,
+          fieldType& testData,
+          arma::mat& testLabel,
+          const double testRatio,
+          const bool shuffleData = true)
+{
+  // Calculate number of objects in the image.
+  size_t numberOfObjects = inputLabel[0].n_rows;
+  arma::mat labelsTemp(numberOfObjects, inputLabel.size());
+
+  for (size_t i = 0; i < inputLabel.size(); i++)
+    labelsTemp.col(i) = inputLabel[i];
+
+  auto completeDataset = arma::join_cols(input, labelsTemp);
+  Split(completeDataset, trainData, testData,
+      testRatio, shuffleData);
```

Comment:
How would we join ```arma::field<mat/vec> ``` to a  ```arma::mat ```?

Reply:
We won't, we will find out which indices belong to training and test set and fill them in respective type using the order vector.